api/specs/webserver/scripts/openapi_projects.py

```diff
@@ -0,0 +1,53 @@
+""" Helper script to automatically generate OAS
+
+This OAS are the source of truth
+"""
+
+# pylint: disable=redefined-outer-name
+# pylint: disable=unused-argument
+# pylint: disable=unused-variable
+# pylint: disable=too-many-arguments
+
+
+from enum import Enum
+
+from _common import (
+    CURRENT_DIR,
+    assert_handler_signature_against_model,
+    create_openapi_specs,
+)
+from fastapi import FastAPI
+from models_library.generics import Envelope
+from models_library.projects import ProjectID
+from models_library.services import UserWithoutServiceAccess
+from simcore_service_webserver.projects.projects_handlers_crud import ProjectPathParams
+
+app = FastAPI(redoc_url=None)
+
+TAGS: list[str | Enum] = [
+    "project",
+]
+
+
+#
+# API entrypoints
+#
+
+
+@app.post(
+    "/projects/{project_id}/shareable",
```

Comment:
for such endpoint we typically use the following conventions
```POST/GET/PATCH/PUT/DELETE /resource``` for a resource
```POST /resource:action``` for an action
in your case shareable is clearly not a resource but an action so it should be ```/projects/{project_id}:shareable```

Reply:
Ok I have a little bit redo the endpoints they are more restful :+1: let me know what you think. 
https://github.com/ITISFoundation/osparc-simcore/pull/4253#issue-1715533321